site/app/views/forum/ForumThreadView.php

```diff
@@ -590,162 +590,146 @@ public function showAlteredDisplayList($threads, $filtering, $thread_id, $catego
-HTML;
-						}
-						if($this->core->getQueries()->isThreadLocked($thread['id']))
-                        {
-                            $return .= <<<HTML
-                            <i class="fas fa-lock" style="padding-left:3px;position:relative; float:right; display:inline-block; color: white; -webkit-text-stroke-width: 1px;
-    -webkit-text-stroke-color: black;" title="Locked" aria-hidden="true"></i>
-HTML;
+            if (!empty($temp_first_post_content)) {
+                $first_post_content = $temp_first_post_content;
+            }
 
-                        }
-						$categories_content = array();
-						foreach ($thread["categories_desc"] as $category_desc) {
-							$categories_content[] = array(htmlentities($category_desc, ENT_QUOTES | ENT_HTML5, 'UTF-8'));
-						}
-						for ($i = 0; $i < count($thread["categories_color"]); $i+=1) {
-							$categories_content[$i][] = $thread["categories_color"][$i];
-						}
-						$return .= <<<HTML
-						<h4>{$titleDisplay}</h4>
-						<h5 style="font-weight: normal;">{$contentDisplay}</h5>
-HTML;
-						foreach ($categories_content as $category_content) {
-							$return .= <<<HTML
-							<span class="label_forum" style="background-color: {$category_content[1]}">{$category_content[0]}</span>
-HTML;
-						}
-						if(!is_null($date)) {
-							$return .= <<<HTML
-							<h5 style="float:right; font-weight:normal;margin-top:5px">{$function_date($date,"n/j g:i A")}</h5>
-HTML;
-						}
-						$return .= <<<HTML
-						</div>
-						</a>
-						<hr style="margin-top: 0px;margin-bottom:0px;">
-HTML;
-					}
-					return $return;
-	}
+            $sizeOfContent = strlen($first_post_content);
+            $contentDisplay = substr($first_post_content, 0, ($sizeOfContent < 80) ? $sizeOfContent : strrpos(substr($first_post_content, 0, 80), " "));
+            $titleLength = strlen($thread['title']);
+
+            $titleDisplay = substr($titleDisplay, 0, ($titleLength < 40) ? $titleLength : strrpos(substr($titleDisplay, 0, 40), " "));
+
+            if (strlen($first_post["content"]) > 80) {
+                $contentDisplay .= "...";
+            }
+            if (strlen($thread["title"]) > 40) {
+                //Fix ... appearing
+                if (empty($titleDisplay))
+                    $titleDisplay .= substr($thread['title'], 0, 30);
+                $titleDisplay .= "...";
+            }
+            $titleDisplay = ($display_thread_ids ? "({$thread['id']}) " : '') . $titleDisplay;
+
+            $link = $this->core->buildUrl(['component' => 'forum', 'page' => 'view_thread', 'thread_id' => $thread['id']]);
+
+            $favorite = isset($thread['favorite']) && $thread['favorite'];
+
+            if (!isset($thread['status'])) {
+                $thread['status'] = 0;
+            }
+            if ($thread['status'] != 0) {
```

Comment:
I don't believe these values should be only initialized when `$thread['status'] != 0`. None of these parameters seem to get set when `== 0` and will throw errors. 


![Screenshot from 2019-06-06 01-50-33](https://user-images.githubusercontent.com/16356240/59009930-8509d300-87fd-11e9-8970-d45fcd8f0a51.png)

Reply:
I kind of just converted the existing logic to use Twig..however, these errors I did not face! How did you come about them? :o

Reply:
These issues occur when the thread status is equal to 0. When publishing a thread you can change the thread status by selecting `Unresolved (-1), Resolved (1), or Comment (0)`.

Reply:
Apparently, there wasn't a problem with the logic, the variables being set there were local, hence producing error (which was not the case with the non-refactored code as they were sending output from within the if blocks).